lib/ruby_lsp/setup_bundler.rb

```diff
@@ -183,22 +194,26 @@ def run_bundle_install(bundle_gemfile = @gemfile)
       local_config_path = File.join(Dir.pwd, ".bundle")
       env["BUNDLE_APP_CONFIG"] = local_config_path if Dir.exist?(local_config_path)
 
-      # If both `ruby-lsp` and `debug` are already in the Gemfile, then we shouldn't try to upgrade them or else we'll
-      # produce undesired source control changes. If the custom bundle was just created and either `ruby-lsp` or `debug`
-      # weren't a part of the Gemfile, then we need to run `bundle install` for the first time to generate the
-      # Gemfile.lock with them included or else Bundler will complain that they're missing. We can only update if the
-      # custom `.ruby-lsp/Gemfile.lock` already exists and includes both gems
+      # If `ruby-lsp` and `debug` (and potentially `ruby-lsp-rails`) are already in the Gemfile, then we shouldn't try
+      # to upgrade them or else we'll produce undesired source control changes. If the custom bundle was just created
+      # and any of `ruby-lsp`, `ruby-lsp-rails` or `debug` weren't a part of the Gemfile, then we need to run `bundle
+      # install` for the first time to generate the Gemfile.lock with them included or else Bundler will complain that
+      # they're missing. We can only update if the custom `.ruby-lsp/Gemfile.lock` already exists and includes all gems
 
       # When not updating, we run `(bundle check || bundle install)`
       # When updating, we run `((bundle check && bundle update ruby-lsp debug) || bundle install)`
       command = +"(bundle check"
 
       if should_bundle_update?
-        # If ruby-lsp or debug are not in the Gemfile, try to update them to the latest version
+        # If any of `ruby-lsp`, `ruby-lsp-rails` or `debug` are not in the Gemfile, try to update them to the latest
+        # version
         command.prepend("(")
         command << " && bundle update "
         command << "ruby-lsp " unless @dependencies["ruby-lsp"]
         command << "debug " unless @dependencies["debug"]
+        if @dependencies["rails"]
+          command << "ruby-lsp-rails " unless @dependencies["ruby-lsp-rails"]
+        end
```

Comment:
Not sure if the line fits, but let's at least merge the conditionals.
```suggestion
          command << "ruby-lsp-rails " if @dependencies["rails"] && !@dependencies["ruby-lsp-rails"]
```